### ISSUE TITLE
poor parsing time of dash manifest on WebOS 

### ISSUE THREAD

**Issue by pawelgliwinski** [google/shaka-player#1640]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**:
Yes
**What version of Shaka Player are you using?**:
2.4.4
**Can you reproduce the issue with our latest release version?**:
yes
**Can you reproduce the issue with the latest code from `master`?**:
didn't check
**Are you using the demo app or your own custom app?**:
custom app
**If custom app, can you reproduce the issue using our demo app?**:
yes
**What browser and OS are you using?**:
I'm using LG WebOS 3.0 (LG lh615v)
**What are the manifest and license server URIs?**:
I can't post it publicly

**What did you do?**

I'm starting the Dash/Widevine content on LG tv.

**What did you expect to happen?**
I was expecting the title to start at around 6 to 7 seconds

**What actually happened?**

The startup time of title took +35 seconds
When starting the title it takes +35 seconds to parse DASH manifest that is of size 1,5 MB. It is seen that most of the time is consumed on recursive call to shaka.dash.DashParser.processXlinks
![title startup](https://user-images.githubusercontent.com/15032231/47354914-4edd7000-d6c0-11e8-8762-6b5be300c64f.jpeg). The title that has 90kb manifest size starts at around 10 seconds which is still quite slow. Other tvs like WebOS 4 based LG suffer from that issue as well.

What is the cause of such poor performance when it comes to parsing the dash manifest? Are there any config params that could speed up the process?

**Comment by TheModMaker** [comment]
Also how did you get that dump?  It looks similar to Chrome performance tab.  Can you save a dump of it and send it to us?  You can also privately send the dump and media URLs to shaka-player-issues@google.com and we'll keep it within the team.

**Comment by pawelgliwinski** [comment]
>  It would really help if you could provide some example content to test with. 
  Not sure if there is a clear content there, I'll try to get something in the meantime I'll send the manifest over the email for start.

>Also how did you get that dump? It looks similar to Chrome performance tab. 
the dump was taken with LG webinspector.

> Does your content actually use xlinks? 
 I don't think we are using xlinks,.

**Comment by pawelgliwinski** [comment]
Hi,

 

I’m testing that now and it gives good results already I think.

 

Pawel

**Comment by joeyparrish** [comment]
Xlink optimizations have been cherry-picked to v2.4.5.

**Comment by chadacious** [comment]
Seems that the processing of xlinks can still be long operation even with this change in shaka-player 3. Attempting to process a very large mpd (3mb) of the form <SegmentList ...><SegmentUrl mediaRange="..-.." /> will actually crash after several minutes stuck in the processXlinks function of mpd_utils.js.

Should there be a config option to disable xlink processing if we know that our mpd doesn't use xlink?

**Comment by avelad** [comment]
The config was added in https://github.com/google/shaka-player/commit/014e14638bb1d2b6bbdbbdcbfe48e0eef821c47c , is it enough?

**Comment by joeyparrish** [comment]
Yes, but there was no "closes" in PR #3240, so the issue didn't get closed automatically.

@chadacious, you can now use `manifest.dash.disableXlinkProcessing` to disable xlink.  Thanks!
